src/libopenrave/kinbodygrab.cpp

```diff
@@ -18,32 +18,51 @@
 
 namespace OpenRAVE {
 
-/// \brief Push link to listNonCollidingLinksWhenGrabbed, only if it has no collision with the whole grabbedBody.
-static void _PushLinkIfNonColliding(std::list<KinBody::LinkConstPtr>& listNonCollidingLinksWhenGrabbed,
+/// \brief Push link to listNonCollidingLinkPairs if the given links are not in collision each other. This is for grabbed-grabber link pairs.
+static void _PushLinkIfNonColliding(std::list<std::pair<KinBody::LinkConstPtr, KinBody::LinkConstPtr> >& listNonCollidingLinkPairs,
```

Comment:
```suggestion
/// \brief Push (grabbedBodyLink, grabberLink) pairs to the given listNonCollidingLinkPairs if they are not colliding
///        with each other. Note that the order of pair will always be such that the first element is the grabbed
///        body's link and the second element is the grabber's link.
static void _PushLinkIfNonColliding(KinBody::ListNonCollidingLinkPairs& listNonCollidingLinkPairs,
                                    CollisionCheckerBasePtr& pchecker,
                                    const KinBody::LinkPtr& pGrabberLinkToCheck, const KinBody& grabbedBody)
```

Reply:
thanks!

since there was conflict with other commit, i manually applied this improvement : https://github.com/rdiankov/openrave/pull/1438/commits/455ed5f9b2d381cefefa5e79543e592feae0b8b9